sqlobject/src/main/java/org/jdbi/v3/sqlobject/Handlers.java

```diff
@@ -43,12 +45,21 @@ private Handlers(Handlers that) {
         factories.addAll(that.factories);
     }
 
+    @Override
+    public void setRegistry(ConfigRegistry registry) {
+        this.registry = registry;
+        factories.forEach(hf -> hf.setRegistry(registry));
```

Comment:
Race condition. Multiple threads could be inspecting a new SQL object at once, and they would step on each other with their own config